python/ray/tune/suggest/suggestion.py

```diff
@@ -91,7 +93,10 @@ def _generate_trials(self, experiment_spec, output_path=""):
                 else:
                     break
             spec = copy.deepcopy(experiment_spec)
-            spec["config"] = suggested_config
+            if self._merge_configs:
+                spec["config"].update(suggested_config)
+            else:
+                spec["config"] = suggested_config
             self._counter += 1
             tag = "{0}_{1}".format(
                 str(self._counter), format_vars(spec["config"]))
```

Comment:
Here, `format_vars` needs a resolved vars as input (https://github.com/ray-project/ray/blob/cf0c4745f4c2953a774bbf2a944c3cdc76bdae24/python/ray/tune/suggest/variant_generator.py#L100).

However, spec["config"] is not resolved, which will lead to unexpected tag name. In my codebase, I work around as follows:
<pre>
            self._counter += 1
            def resolve(spec):
                res = {}
                for k, v in spec.items():
                    if isinstance(v, dict):
                        for k_, v_ in resolve(v).items():
                            res[(k, ) + k_] = v_
                    else:
                        res[(k, )] = v
                return res
            resolved_config = resolve({"config": suggested_config})
            tag = "{0}_{1}".format(
                str(self._counter), format_vars(resolved_config))
</pre>

Reply:
What do you use the tag for? Why not use the `trial_name_creator`?

Reply:
This problem was found at the moment the `trial_name_creator` was not introduced. Correcting this could provide better TensorBoard visualization (e.g., filter plots by folder names) by default for those who do not want extra codes for `trial_name_creator`.

Reply:
OK added.